src/pages/fundamentals/eutxo-patterns.mdx

```diff
@@ -29,23 +29,231 @@ the transaction contain the corresponding `Input`. By doing this, the minting po
 validate a single time. In some designs, this logic is used for only a subset of redeemers to allow more
 flexible minting policies.
 
+
+#### NFT example:
+An NFT (Non-Fungible Token) must be created using a one-shot minting policy that ensures each minted value
+is validated by spending a specific UTxO provided through the transaction inputs. This minting policy uses
+a validator parameter of OutputReference to confirm that the transaction spends the UTxO. Additionally, the 
+policy guarantees that only one token is minted, ensuring the NFT's uniqueness.
+
+To create an NFT minting policy, first define OutputReference and AssetName as parameters and set the action
+type to mint or burn the token based on the value provided in the redeemer.
+
+``` aiken
+type Action {
+  Minting
+  Burning
+}
+
+validator(utxo_ref: OutputReference, token_name: AssetName) {
+  fn nft(action: Action, ctx: ScriptContext) -> Bool {
+      todo @"mint and burn"      
+    }
+  }
+}
+```
+
+The validator expects a minting script purpose and ensures that only one value is minted; it will fail otherwise.
+
+``` aiken
+validator(utxo_ref: OutputReference, token_name: AssetName) {
```

Comment:
We don't strictly need to check what is minted, the primary pattern we are trying to demonstrate here is generating uniqueness with a one-shot policy. `token_name` is unnecessary.